embassy-sync/src/signal.rs

```diff
@@ -28,8 +31,11 @@ use core::task::{Context, Poll, Waker};
 ///
 /// static SOME_SIGNAL: Signal<SomeCommand> = Signal::new();
 /// ```
-pub struct Signal<T> {
-    state: UnsafeCell<State<T>>,
+pub struct Signal<T, R = CriticalSectionRawMutex>
+where
+    R: RawMutex,
+{
+    state: Mutex<R, RefCell<State<T>>>,
```

Comment:
I think it should be possible to use `Cell` instead of `RefCell`.

Reply:
Let me try.

Reply:
Yep, works fine.